packages/react/src/themes/teams/components/Button/buttonStyles.ts

```diff
@@ -41,30 +40,73 @@ const buttonStyles: ComponentSlotStylesInput<ButtonProps, ButtonVariables> = {
       }),
 
       // rectangular button defaults
-      ...(!p.text && {
-        outline: 0,
-        borderWidth,
-        borderStyle: 'solid',
-        borderColor: v.borderColor,
-        boxShadow: v.boxShadow,
+      ...(!p.circular &&
+        !p.text && {
+          borderStyle: 'solid',
+          borderColor: v.borderColor,
+          boxShadow: v.boxShadow,
+          borderRadius: '.2rem',
+          border: 0,
 
-        ':hover': {
-          color: v.colorHover,
-          backgroundColor: v.backgroundColorHover,
-          borderColor: v.borderColorHover,
-        },
+          '::after': {
+            position: 'absolute',
+            content: '""',
+            borderRadius: '.2rem',
+            border: '1px solid',
+            borderColor: '#e1dfdd',
+            top: 0,
+            right: 0,
+            bottom: 0,
+            left: 0,
+            ...(p.primary && {
+              borderColor: v.primaryBorderColor,
+            }),
+          },
 
-        ':focus': {
-          ...borderFocusStyles[':focus'],
-          boxShadow: 'none',
-          ':active': { backgroundColor: v.backgroundColorActive },
-        },
-        ':focus-visible': {
-          ...borderFocusStyles[':focus-visible'],
-          color: v.colorFocus,
-          backgroundColor: v.backgroundColorFocus,
-        },
-      }),
+          '::before': {
+            position: 'absolute',
+            content: '""',
+            borderRadius: '.2rem',
+            border: '.1rem solid transparent',
+            top: '.1rem',
+            right: '.1rem',
+            bottom: '.1rem',
+            left: '.1rem',
+          },
+
+          ...(p.isFromKeyboard && {
```

Comment:
`isFromKeyboard` does not exists any more, use the":focus-visible" instead. Also, never use multiple pseudo selectors at the same time. You can change the following lines to:
```
":focus": {
  ":active": {
    borderColor: '...',
  },
}
```